project/Scripts/UnitTests.gd

```diff
@@ -0,0 +1,95 @@
+extends Node
+
+#Test addNode
+func test_addNode():
+	var graph = ProofGraph.new()
+	for i in range(0, 50, 1):
+		var nodeInfo = {"location": [i, i+1, i+2], "node_id": i}
+		graph.addNode(nodeInfo)
```

Comment:
These tests aren't robust enough.

There is nothing special about 2-50 that isn't already being tested by 1.
Unit tests should focus on all types of inputs and edge cases: weird situations where the code may break, but needs to be resilient.

This should minimally test 
1) negative locations
2) floating point locations
3) massive locations (around the integer/float limit)
4) illegal locations: IE: Over the integer limit, Special floating values: NaN, Inf, -Inf. and fail gracefully. 



and show it is reliant against them.